### ISSUE TITLE
[BUG] Last page is zeroed unless a workaround is applied

### ISSUE THREAD

**Issue by iron76** [KWARC/LaTeX-proposal#15]
This is related to #14. I tried the latest `LaTeX-proposal` code (cloned from github) and the problem persists. The last page is labelled 0. I found out a workaround consisting in writing something in between `\end{proposal}` and `\end{document}`. This was actually why initially I was assuming the problem related to the finalised version (see my previous issue https://github.com/pdebuyl/FET_Proposal/issues/1).

**Comment by kohlhase** [comment]
I still cannot reproduce your problem, and I see that https://github.com/pdebuyl/FET_Proposal/tree/master/LaTeX-proposal is still not updated. But I also do not see what you are applying it to. Can you either
* make a minimal example that shows this behavior or 
* make sure that you are running the right proposal.cls.

**Comment by iron76** [comment]
Sorry. I forgot to say that the bug is replicable with one of the examples in your repository (please notice that this bug has nothing to do with the repository by @pdebuyl, otherwise I would have opened the issue there). Here are the steps to generate a .pdf with zeroed last page:
```
git clone https://github.com/KWARC/LaTeX-proposal.git 
cd LaTeX-proposal/eu/examples/fetopenstrep/
pdflatex propB.tex 
... (repeat as many times as you want)
pdflatex propB.tex 
```
The produced `propB.pdf ` has tags `page n of 0` in all pages.

**Comment by pdebuyl** [comment]
Using pdflatex instead of the Makefile results in your system-wide copy of the style files to be used instead of those of the repository. Check the propB.log file to check which files are used.

It is possible to set the TEXINPUTS environment variables, see https://github.com/pdebuyl/FET_Proposal/blob/master/debian-env.sh

**Comment by iron76** [comment]
I checked and I am using the correct style files. Further digging I found out that the issue is instead related to the fact that I am not compiling the bibliography. As soon I do it the last page is correctly set. The issue is crucial in my proposal because I don't have a bibliography.

**Comment by kohlhase** [comment]
Thanks for digging, I can reproduce now, and I am quite sure that I can fix this. 
The main problem is that I never envisioned that someone would not want references. I will probably have to make this configurable.

**Comment by pdebuyl** [comment]
Indeed, a single bibliography element solves the problem. This is thus a rather uncommon problem.

Also, in the case of an empty bibliography, biblatex throws a "latex warning". Is that the cause of the missing {prop}{pages}{last} pdata reference?

**Comment by kohlhase** [comment]
It seems that this is a timing issue: LaTeXML closes the pdata file before it has the chance to write the last page. We probably have to build in something that flushes the buffers (and opening another file does that I think). Or we have to make the ```\protected@write``` I am using ```\immediate``` (I thought it was). 
I have to investigate further, but do not currently have the time.
